src/Rings/mpoly-graded.jl

```diff
@@ -1573,18 +1569,16 @@ end
 
 function Oscar.degree(H::HilbertData)
 
-  for i = 1:ngens(base_ring(H.I))
-     if H.weights[i] != 1
-       throw(ArgumentError("All weights must be 1."))
-     end
-  end
+  all(isone, H.weights) || throw(ArgumentError("All weights must be 1"))
   
   P = hilbert_polynomial(H)
-  if P==zero(parent(P))
+  if iszero(P)
      q, _ = hilbert_series(H, 2)
      return q(1)
   end
-  return leading_coefficient(P)*factorial(degree(P))
+  deg = leading_coefficient(P)*factorial(ZZ(degree(P)))
+  @assert isone(denominator(deg))
+  return numerator(deg)
```

Comment:
@wdecker please verify this is correct

Reply:
I am not wdecker, but the assertion is correct